src/vstart.sh

```diff
@@ -1232,5 +1232,17 @@ if [ "$CEPH_DIR" != "$PWD" ]; then
     echo "export CEPH_KEYRING=$keyring_fn"
 fi
 
+# convenience wrapper for running cephfs-shell out of build dir
+if [ $MDS -gt 0  -a -n "$CEPH_ROOT" ]; then
+	cat > $CEPH_BIN/vstart-cephfs-shell << EOF
+#!/bin/bash
+
+export LD_LIBRARY_PATH=$CEPH_LIB
+export PYTHONPATH=$PYBIND:$CEPH_LIB/cython_modules/lib.$CEPH_PY_VERSION_MAJOR
```

Comment:
What I'd prefer to see here is to fix the echo'd exports above this line. I understand the difficulty is that most people don't source `vstart.sh`. Also, there's other clutter in vstart's stdout.

So perhaps let's also change the code to `tee` the environment to a file that can be sourced later, like `build/vstart_environment.sh`. Then, we can also adjust `cephfs-shell` to complain if it cannot import `rados` or `cephfs`. Maybe add some specific detection for vstart environments (i.e. does `./vstart_environment.sh` exist) so we can give a helpful error message.

Reply:
> Also, the missing executable bits on `cephfs-shell` should be backported I think. Please create a tracker for that.

What problem are we solving by backporting this? I only changed it as a convenience for running it out of the tree. The debian cephfs-shell package seems to install it with the right mode.

Reply:
Oh, then nevermind.

Reply:
Ok, that works. I had it just tee those values to a new vstart_environment.sh file. If you source that into your shell then cephfs-shell can be run directly.